cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatabaseImpl.java

```diff
@@ -167,30 +167,22 @@
      */
     private final AttachmentStreamFactory attachmentStreamFactory;
 
-    protected DatabaseImpl(String dir, String name) throws SQLException, IOException,
-            DatastoreException {
-        this(dir, name, new NullKeyProvider());
-    }
-
     /**
      * Constructor for single thread SQLCipher-based datastore.
      * @param dir The directory where the datastore will be created
-     * @param name The user-defined name of the datastore
      * @param provider The key provider object that contains the user-defined SQLCipher key
      * @throws SQLException
      * @throws IOException
      */
-    protected DatabaseImpl(String dir, String name, KeyProvider provider) throws SQLException,
+    DatabaseImpl(File dir, KeyProvider provider) throws SQLException,
```

Comment:
I think this should be private, or it can probably stay package if it moves into an internal package. I realise a bunch of tests are calling `new DatabaseImpl` but really they should be migrated where possible (which I think should be everywhere) to use the `getInstance().database` API (and I guess casting to `DatabaseImpl` in some cases).